hasher-matcher-actioner/scripts/submit_content_test.py

```diff
@@ -0,0 +1,165 @@
+            )
+            exit(1)
+
+        self.api = HasherMatcherActionerAPI(
+            api_url, api_token, client_id, refresh_token
+        )
+
+    def _refresh_token(self):
+        self.api._refresh_token()
+
+    def create_dataset_config(
+        self,
+        privacy_group_id: str,
+        privacy_group_name: str,
+        description: str = "",
+        matcher_active: bool = True,
+        fetcher_active: bool = False,
+        write_back: bool = False,
+    ):
+        self.api.create_dataset_config(
+            privacy_group_id,
+            privacy_group_name,
+            description,
+            matcher_active,
+            fetcher_active,
+            write_back,
+        )
+
+    def create_action(
+        self,
+        action_performer: ActionPerformer,
+    ):
+        self.api.create_action(
+            name=action_performer.name,
+            config_subtype=action_performer.get_config_subtype(),
+            fields={
+                key: value
+                for key, value in vars(action_performer).items()
+                if key not in {"name", "config_subtype"}
+            },
+        )
+
+    def create_action_rule(
+        self,
+        action_rule: ActionRule,
+    ):
+        # Need to give the api a json like dict object (just like is used in aws)
+        self.api.create_action_rule(action_rule.to_aws())
+
+
+def set_up(helper: DeployedInstanceTestHelper):
+    helper.create_dataset_config(
+        privacy_group_id=helper.SAMPLE_PG_ID,
+        privacy_group_name="Holiday Sample Set",
+    )
+
+    action_performer = WebhookGetActionPerformer(
+        name="TestActionWebhookGet",
+        url="not_a_real_url",
+        headers='{"this-is-a":"test-header"}',
+    )
+
+    helper.create_action(
+        action_performer=action_performer,
+    )
+
+    action_rule = ActionRule(
+        name="Trigger for holidays_jpg1_dataset tag 2",
+        action_label=ActionLabel("TestActionWebhookGet"),
+        must_have_labels=set(
+            [
+                ClassificationLabel("holidays_jpg1_dataset"),
+            ]
+        ),
+        must_not_have_labels=set(),
+    )
+
+    helper.create_action_rule(
+        action_rule=action_rule,
+    )
+
+
+def submit_content(helper: DeployedInstanceTestHelper):
+    content_id = "submit_content_test_id_1"
+    filepath = "sample_data/b.jpg"
+    additional_fields = ["i-am:a-bridge", "this-is:a-test"]
+    with open(filepath, "rb") as file:
+        helper.api.send_single_submission_b64(
+            content_id,
+            file,
+            additional_fields,
+        )
+
+
+def check_for_resulting_objects(helper: DeployedInstanceTestHelper):
+    # TODO impl
+    pass
+
+
+if __name__ == "__main__":
+    # If you want hard code tests for methods you can do so here:
+
+    # Add init values for the api:
+    # `os.environ.get(...)` will enable us pass values from `terraform output` in sh
+
+    HMA_TOKEN = os.environ.get("HMA_TOKEN", "")
```

Comment:
nit: Why do these look like constants?

Reply:
because they were, this is matching the style in our lambdas. Happy to change it if you'd prefer. 

https://github.com/facebook/ThreatExchange/blob/master/hasher-matcher-actioner/hmalib/lambdas/api/api_root.py#L32-L35

Reply:
changed them, thanks for flagging.